arch/arm64/src/rk3399/hardware/mx8mp_gpio.h

```diff
@@ -0,0 +1,87 @@
+/****************************************************************************
+ * arch/arm/src/mx8mp/hardware/mx8mp_gpio.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/* Reference:
+ *   "i.MX 8M Plus Applications Processor Reference Manual",
+ *   Document Number: IMX8MPRM Rev. 1, 06/2021. NXP
+ */
+
+#ifndef __ARCH_ARM_SRC_MX8MP_HARDWARE_MX8MP_GPIO_H
+#define __ARCH_ARM_SRC_MX8MP_HARDWARE_MX8MP_GPIO_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include "hardware/rk3399_memorymap.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define GPIO_NPORTS               5      /* Five total ports */
+#define GPIO_NPINS                32     /* Up to 32 pins per port */
+
+/* GPIO Register Offsets ****************************************************/
+
+#define DR_OFFSET                 0x0000  /* Data Register */
+#define GDIR_OFFSET               0x0004  /* Data Direction Register */
+#define PSR_OFFSET                0x0050  /* GPIO External Port Register */
+#define ICR1_OFFSET               0x000c  /* Interrupt Configuration Register 1 */
+#define ICR2_OFFSET               0x0010  /* Interrupt Configuration Register 2 */
+#define IMR_OFFSET                0x0014  /* Interrupt Mask Register */
+#define ISR_OFFSET                0x0018  /* Interrupt Status Register */
+#define EDGE_OFFSET               0x001c  /* Interrupt Status Register */
+
+/* GPIO Register Addresses **************************************************/
+//if gpio2 add + 6 0000 due to different base address
+//TODO: should we do this as a inline function 
+#define GPIO_BASE_ADDR(n)            ((2 >= n ) ? RK3399_GPIO0_ADDR : RK3399_GPIO2_ADDR)
```

Comment:
@lupyuen
This is the best Idea I got for being able to switch the base address between the PMUGRF and GRF since their gpio have different base addresses. 
I have todo this also on the iomux file but was wondering what suggestions you had that could be better.